code/go/0chain.net/smartcontract/dbs/event/blobber.go

```diff
@@ -0,0 +1,162 @@
+package event
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"0chain.net/smartcontract/dbs"
+
+	"gorm.io/gorm"
+)
+
+type Blobber struct {
+	gorm.Model
+	BlobberID string `json:"id" gorm:"uniqueIndex"`
+	BaseURL   string `json:"url"`
+
+	// geolocation
+	Latitude  float64 `json:"latitude"`
+	Longitude float64 `json:"longitude"`
+
+	// terms
+	ReadPrice               int64   `json:"read_price"`
+	WritePrice              int64   `json:"write_price"`
+	MinLockDemand           float64 `json:"min_lock_demand"`
+	MaxOfferDuration        string  `json:"max_offer_duration"`
+	ChallengeCompletionTime string  `json:"challenge_completion_time"`
+
+	Capacity        int64 `json:"capacity"` // total blobber capacity
+	Used            int64 `json:"used"`     // allocated capacity
+	LastHealthCheck int64 `json:"last_health_check"`
+
+	// stake_pool_settings
+	DelegateWallet string  `json:"delegate_wallet"`
+	MinStake       int64   `json:"min_stake"`
+	MaxStake       int64   `json:"max_stake"`
+	NumDelegates   int     `json:"num_delegates"`
+	ServiceCharge  float64 `json:"service_charge"`
+}
+
+func (bl *Blobber) exists(edb *EventDb) (bool, error) {
+	var count int64
+	result := edb.Get().
+		Model(&Blobber{}).
+		Where("blobber_id", bl.BlobberID).
+		Count(&count)
+	if result.Error != nil {
+		return false, fmt.Errorf("error searching for blobber %v, error %v",
+			bl.BlobberID, result.Error)
+	}
+	return count > 0, nil
+}
+
+func (bl *Blobber) create(edb *EventDb) error {
+	result := edb.Store.Get().Create(bl)
+	return result.Error
+}
+
+func (edb *EventDb) GetBlobber(id string) (*Blobber, error) {
+	exists, err := (&Blobber{
+		BlobberID: id,
+	}).exists(edb)
+	if err != nil {
+		return nil, err
+	}
+	if !exists {
+		return nil, fmt.Errorf("blobber %v not in event db", id)
+	}
+
+	var blobber Blobber
+	result := edb.Store.Get().
+		Model(&Blobber{}).
+		Find(&Blobber{}).
```

Comment:
I think you can avoid using Find and just use First instead. Also I am not sure how it will unwrap Blobber object, in where you adding blobber_id, but Blobber doesn't have blobber_id field

Reply:
Blobber has a `BlobberID` field, and gorm seems to convert that to `blobber_id`, `BlobberID` would probably also work. I should be consistent here, for example [overwriteBlobber](https://github.com/0chain/0chain/blob/blobber-events/code/go/0chain.net/smartcontract/dbs/event/blobber.go#L100).